cmd/snap-confine/mount-support.c

```diff
@@ -429,6 +429,15 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 			    SC_HOSTFS_DIR);
 		}
 	}
+	// Ensure that hostfs isgroup owned by root. We may have (now or earlier)
+	// created the directory as the user who first ran a snap on a given
+	// system and the group identity of that user is visilbe on disk.
+	// This was LP:#1665004
+	if (chown(SC_HOSTFS_DIR, 0, 0) < 0) {
+		die("cannot change user/group owner of %s to root",
+		    SC_HOSTFS_DIR);
+	};
```

Comment:
FYI, please remove the trailing ';' here.

The cause of this is because we dropped euid and egid earlier, then by the time this code is run, we had raised euid to do a few things (but not gid, cause it wasn't strictly needed). We could raise gid too, but I prefer this approach. I was expecting this to be directly after the mkdir() call. As written, chown will *always* be called, forever and ever on every launch of a command. In a perfect world we would have done this chown when mkdir() was added, but there are systems out there that need to be fixed so I understand why this is where it is. Can we wrap this in a stat() call? Eg:
```
struct stat sb;
if (stat(SC_HOSTFS_DIR), &sb) == -1) {
    die("cannot stat() %s", SC_HOSTFS_DIR);
}
if (sb.st_uid != 0 || sb.st_gid != 0) {
    if (chown(SC_HOSTFS_DIR, 0, 0) < 0) {
        die("cannot change user/group owner of %s to root", SC_HOSTFS_DIR);
    }
}